pkg/credentialprovider/azure/azure_credentials.go

```diff
@@ -186,6 +191,30 @@ func (a *acrProvider) Enabled() bool {
 }
 
 func (a *acrProvider) Provide(image string) credentialprovider.DockerConfig {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+
+	// If the cache hasn't expired, return our cache
+	if time.Now().Before(a.expiration) {
+		if a.config != nil && a.config.UseManagedIdentityExtension {
+			loginServer := a.parseACRLoginServerFromImage(image)
+			if loginServer == "" {
+				return a.cacheDockerConfig
```

Comment:
This sort of behavior is not something the externalized API will cover, if we can't determine the registry (login server), we should log an error and return an empty `credntialprovider.DockerConfig`

Reply:
This would also result in the ACR provider returning cached docker config for non-ACR repos?

Reply:
fixed in line 203, PTAL.